src/shared/edit-util.h

```diff
@@ -5,27 +5,30 @@
 
 typedef struct EditFile {
         char *path;
-        char *tmp;
+        char *original_path;
+        char **comment_paths;
+        char *temp;
         unsigned line;
 } EditFile;
 
-#define _EDIT_FILE_FOREACH(s, e, i)                                     \
-        for (typeof(*(e)) *s, *i = (e); (s = i) && i->path; i++)
+typedef struct EditFileContext {
+        EditFile *files;
+        size_t n_files;
+        const char *marker_start;
+        const char *marker_end;
+} EditFileContext;
 
-#define EDIT_FILE_FOREACH(s, e)                                         \
-        _EDIT_FILE_FOREACH(s, e, UNIQ_T(i, UNIQ))
+#define EDIT_FILES_FOREACH(s, c)                                        \
+        for (EditFile *s = (c).files; s < (c).files + (c).n_files; s++)
```

Comment:
Hmm... This evaluates `c` multiple times... I really do not like the macro.
At least, please make the macro takes `EditFileContext*` rather than `EditFileContext`.